sw/device/tests/usbdev_mixed_test.c

```diff
@@ -0,0 +1,308 @@
+ * DPI model and will increase the simulation time)
+ */
+static bool verbose = false;
+
+/*
+ * These switches may be modified manually to experimentally measure the
+ * performance of IN traffic in isolation, or IN/OUT together etc.
+ *
+ * Are we sending data?
+ */
+static bool sending = true;
+
+/**
+ * Are we generating a valid byte sequence?
+ */
+static bool generating = true;
+
+/**
+ * Do we want the host to retry transmissions? (DPI model only; we cannot
+ * instruct a physical host to fake delivery failure/packet corruption etc)
+ */
+static bool retrying = true;
+
+/**
+ * Are we expecting to receive data?
+ */
+static bool recving = true;
+
+/**
+ * Send only maximal length packets?
+ * (important for performance measurements on the USB, but obviously undesirable
+ *  for testing reliability/function)
+ */
+static bool max_packets = false;
+
+/**
+ * Number of streams to be created
+ */
+static const unsigned nstreams = NUM_STREAMS;
+
+OTTF_DEFINE_TEST_CONFIG();
+
+bool test_main(void) {
+  // Context state for streaming test
+  usb_testutils_streams_ctx_t *ctx = &stream_test;
+
+  CHECK(kDeviceType == kDeviceSimDV || kDeviceType == kDeviceSimVerilator,
+        "This test is not expected to run on FPGA platforms at present. It "
+        "needs logic on the host side to support non-Bulk Transfer streams.");
+
+  LOG_INFO("Running USBDEV MIXED Test");
+
+  // Check we can support the requested number of streams
+  CHECK(nstreams && nstreams < USBDEV_NUM_ENDPOINTS);
+
+  // Decide upon the number of bytes to be transferred for the entire test
+  uint32_t transfer_bytes = TRANSFER_BYTES_FPGA;
+  if (kDeviceType == kDeviceSimVerilator) {
+    transfer_bytes = TRANSFER_BYTES_VERILATOR;
+  }
+  transfer_bytes = (transfer_bytes + nstreams - 1) / nstreams;
+  LOG_INFO(" - %u stream(s), 0x%x bytes each", nstreams, transfer_bytes);
+
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  pinmux_testutils_init(&pinmux);
+  CHECK_DIF_OK(dif_pinmux_input_select(
+      &pinmux, kTopEarlgreyPinmuxPeripheralInUsbdevSense,
+      kTopEarlgreyPinmuxInselIoc7));
+
+  // Construct the test/stream flags to be used
+  test_flags = (sending ? kUsbdevStreamFlagRetrieve : 0U) |
+               (generating ? kUsbdevStreamFlagCheck : 0U) |
+               (retrying ? kUsbdevStreamFlagRetry : 0U) |
+               (recving ? kUsbdevStreamFlagSend : 0U) |
+               // Note: the 'max_packets' test flag is not required by the DPI
+               (max_packets ? kUsbdevStreamFlagMaxPackets : 0U);
+
+  // Remember context state for usb_testutils context
+  ctx->usbdev = &usbdev;
+
+  // Call `usbdev_init` here so that DPI will not start until the
+  // simulation has finished all of the printing, which takes a while
+  // if `--trace` was passed in.
+  CHECK_STATUS_OK(usb_testutils_init(ctx->usbdev, /*pinflip=*/false,
+                                     /*en_diff_rcvr=*/false,
```

Comment:
This will need to be `true` for the ASIC, so you might want to consider exporting a symbol to a `volatile` variable that the simulator can override.